skrub/_table_vectorizer.py

```diff
@@ -536,6 +542,11 @@ def _apply_cast(self, X: pd.DataFrame) -> pd.DataFrame:
                 X[col].fillna(value=np.nan, inplace=True)
         for col in self.imputed_columns_:
             X[col] = _replace_missing_in_cat_col(X[col])
+        # If mixed types, convert to string
+        for col in X.columns:
+            unique_types = X[col].apply(lambda x: type(x)).nunique()
+            if unique_types > 1:
+                X[col] = np.where(X[col].isna(), X[col], X[col].astype(str))
```

Comment:
It seems that we are computing this twice, no?

Reply:
We do it both in auto_cast (during fit) and apply_cast (during transform), because for string columns, the type infered during auto_cast is not necessarly string, but can be object because of missing values

Reply:
:+1: